src/main/java/org/sagebionetworks/bridge/services/OAuthProviderService.java

```diff
@@ -136,6 +177,72 @@ protected DateTime getDateTime() {
         }
     }
 
+    /**
+     * Authenticate a Bridge user via the Synapse OAuth server. This is the second contact of the Synapse
+     * server, after a client has sent a user to authenticate on the Synapse server web site and the Synapse 
+     * server has redirected back to the Bridge client with an authentication code. We now exchange it for 
+     * an authentication token, returned with the userid. The information required is very similar to an 
+     * OAuth provider as configured by a study administrator, but it's internal to our system and so it is 
+     * configured like other components of the server software.
+     *   
+     * @param authToken that was passed from the Synapse server back to the authenticating client
+     * @return synapseUserId if the exchange is successful
+     * 
+     * @throws BadRequestException
+     */
+    public String oauthSignIn(OAuthAuthorizationToken authToken) {
```

Comment:
We pass in a vendor ID so in theory, this could be used with another 3rd party provider like Google. (At that point though, we'll have to lookup the configuration using the vendor ID.)

Reply:
Okay the method how returns an AccountId. In the future, a different vendor ID could be used to retrieve some other identifier besides the Synapse ID (email is a logical choice), and an AccountId can be returned with those credentials. So the method is better encapsulated for further vendors, and I've updated the documentation to reflect this.